packages/driver/src/cy/commands/actions/select.ts

```diff
@@ -104,13 +103,16 @@ export default (Commands, Cypress, cy) => {
         $errUtils.throwErrByPath('select.invalid_multiple')
       }
 
+      const subjectChain = cy.subjectChain()
+
       const getOptions = () => {
+        options.$el = cy.getSubjectFromChain(subjectChain)
         let notAllUniqueValues
 
         // throw if <select> is disabled
-        if (!options.force && options.$el.prop('disabled')) {
-          node = $dom.stringify(options.$el)
-          $errUtils.throwErrByPath('select.disabled', { args: { node } })
+        if (!options.force) {
+          cy.ensureElement(options.$el, 'select', options._log)
+          cy.ensureNotDisabled(options.$el, options._log)
```

Comment:
Standardizing error messages - .select() now uses the same `ensures` functions as other commands, since its original error message wasn't meaningfully different (had less information than the default).

It was also missing an 'attached to DOM' check entirely.